pkg/block/fetcher_test.go

```diff
@@ -393,3 +401,271 @@ func TestTimePartitionMetaFilter_Filter(t *testing.T) {
 	testutil.Equals(t, expected, input)
 
 }
+
+func TestCompactedBlockFilter_Filter(t *testing.T) {
+	for _, tcase := range []struct {
+		name     string
+		input    map[ulid.ULID]*metadata.Meta
+		expected func(input map[ulid.ULID]*metadata.Meta) map[ulid.ULID]*metadata.Meta
+	}{
+		{
+			name: "3 non compacted metas in bucket",
```

Comment:
It does not matter if they are really compacted .. It still does not say to me if it's overlapping or not